marklogic-client-api/src/main/java/com/marklogic/client/impl/OkHttpServices.java

```diff
@@ -1244,25 +1242,29 @@ public DocumentDescriptor head(RequestLogger reqlog, String uri,
   public boolean exists(String uri) throws ForbiddenUserException, FailedRequestException {
     return headImpl(null, uri, null, setupRequest(uri, null)) == null ? false : true;
   }
-
+  
+  @Override
+  public ConnectionResult checkConnection() {
+	  Request.Builder request = new Request.Builder()
+		        .url(this.baseUri);
+	  Response response = headImplExec(null, this.baseUri.uri().toString(), null, request);
+	  ConnectionResult connectionResult = new ConnectionResult();
+	  int statusCode = response.code();
+	  if(statusCode < 300) {
+		  connectionResult.setConnected(true);
+	      connectionResult.setStatusCode(statusCode);
```

Comment:
In the Node.js case, we only set the status code in the false case.  Can we take the same approach here by representing the status code in the field and methods of the ConnectionResult as an Integer?